roles/rsyslog/tasks/set_certs.yml

```diff
@@ -11,7 +11,9 @@
         mode: '0444'
       with_items:
         - '{{ __rsyslog_cert_subject }}'
-      when: item.ca_cert_src | d()
+      when:
+        - item.ca_cert_src | d()
+        - (item.tls is defined) | ternary(item.tls, item.use_cert | d(true))
```

Comment:
yes - including the conditions on the Check certs tasks

Reply:
looking again - I think we cannot do this because of this condition - https://github.com/linux-system-roles/logging/blob/43cbb439a25e6a268d1e9498f58408c05b4c1da1/roles/rsyslog/tasks/set_certs.yml#L66

Reply:
Yep, please take a look... https://github.com/avlitman/logging/pull/3

Reply:
@nhosoi @richm 
I checked the pr in my Env and worked with, and without certificates.

Is it ok that the last task "Check certs - key/certs data are provided, but tls is false" is now not a part of the block   ?

Reply:
and also this condition:   
when: __rsyslog_cert_subject | d([])

Reply:
Hi @nhosoi and @richm,
I tested  the pr with (item.tls is defined) | ternary(item.tls, item.use_cert | d(true)) in every task, and for the outer block and the results are the same:
tls is true, key+certs are provided: copy tasks ok, and check tasks skipped.
tls is true and certs+key undefined: copy tasks failed.
tls is true and certs+key empty and the: copy tasks skipped, and the first check task failed.
tls is false and certs+key undefined: all tasks skipped.
 
So the pr lgtm, but I noticed that the last task does not behave as expected:

I tested "Check certs - key/certs data are provided, but tls is false", with (item.tls is defined) | ternary(item.tls, item.use_cert | d(true)) in every task and for the outer block, and the results are the same:
tls is false and certs+key defined: task was skipped.
tls is false and certs+key empty: task was skipped.

Reply:
> I tested "Check certs - key/certs data are provided, but tls is false", with (item.tls is defined) | ternary(item.tls, item.use_cert | d(true)) in every task and for the outer block, and the results are the same:
> tls is false and certs+key defined: task was skipped.
> tls is false and certs+key empty: task was skipped.

Thank you, @avlitman! This result is wrong.
> tls is false and certs+key defined: task was skipped.

Please take a look at https://github.com/avlitman/logging/pull/4.

Reply:
Hi @nhosoi 
avlitman#4 is verified:
I tested "Check certs - key/certs data are provided, but tls is false", with (item.tls is defined) | ternary(item.tls, item.use_cert | d(true)) in every task and for the outer block, and the results are the same:
tls is false and certs+key defined: task failed.
tls is false and some of the certs/key defined: task failed.
tls is false and certs+key empty: task was skipped.
tls is true and cert+key defined: task was skipped.